src/posix/http_client_curl.cc

```diff
@@ -72,22 +84,32 @@ bool HTTPClientPlatformWrapper::RunHTTPRequest() {
   }
 
   cmd += url_requested_;
-  server_response_ = RunCurl(cmd);
-
-  // Clean up tmp file if any was created.
-  if (!post_body_.empty() && !post_file_.empty()) {
-    ::remove(post_file_.c_str());
+  try {
+    server_response_ = RunCurl(cmd);
+
+    // Clean up tmp file if any was created.
+    if (!post_body_.empty() && !post_file_.empty()) {
+      ::remove(post_file_.c_str());
+    }
+
+    error_code_ = -1;
+    std::string & s = server_response_;
+    if (s.size() < 3) {
```

Comment:
These lines look like an aggressive hack to me. Why not do it properly: cut out the first line, `Split` the string, get its first components?